packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -30,15 +30,65 @@ const double _kForegroundScreenOpacityFraction = 0.7;
 ///    the iOS design specific chrome.
 ///  * <https://developer.apple.com/ios/human-interface-guidelines/controls/pickers/>
 class CupertinoPicker extends StatefulWidget {
-  /// Creates a control used for selecting values.
+  /// Creates a picker from a concrete list of children.
   ///
   /// The [diameterRatio] and [itemExtent] arguments must not be null. The
   /// [itemExtent] must be greater than zero.
   ///
   /// The [backgroundColor] defaults to light gray. It can be set to null to
   /// disable the background painting entirely; this is mildly more efficient
   /// than using [Colors.transparent].
-  const CupertinoPicker({
+  ///
+  /// The [looping] argument decides whether the child list loops and can be
+  /// scrolled infinitely.  If set to true scrolling past the end of the list
+  /// will loop the list back to the beginning.  If set to false, the list will
+  /// stop scrolling when you reach the end or the beginning.
+  CupertinoPicker({
+    Key key,
+    this.diameterRatio = _kDefaultDiameterRatio,
+    this.backgroundColor = _kDefaultBackground,
+    this.offAxisFraction = 0.0,
+    this.useMagnifier = false,
+    this.magnification = 1.0,
+    this.scrollController,
+    @required this.itemExtent,
+    @required this.onSelectedItemChanged,
+    @required List<Widget> children,
+    bool looping = false,
+  }) : assert(children != null),
+       assert(diameterRatio != null),
+       assert(diameterRatio > 0.0, RenderListWheelViewport.diameterRatioZeroMessage),
+       assert(magnification > 0),
+       assert(itemExtent != null),
+       assert(itemExtent > 0),
+       childDelegate = looping
+                       ? new ListWheelChildLoopingListDelegate(children: children)
+                       : new ListWheelChildListDelegate(children: children),
+       super(key: key);
+
+  /// Creates a picker from an indexed builder where its children are
+  /// dynamically built during layout.
+  ///
+  /// A child is lazily created when it starts becoming visible in the viewport.
+  /// All of the children provided by the builder are cached and reused, so
+  /// normally the builder is only called once for each index (except when
+  /// rebuilding - the cache is cleared). Therefore, the builder has to be
```

Comment:
Why does it have to return the same widget for the same index? I think the explanation above is fine. If the same builder instance actually returns something different each time, as long as the user understands the implication that it won't be called a second time for it to have an opportunity to return something different, it's fine.

Reply:
It doesn't have to. I'm just assuming so, otherwise, caching is not possible.
We can choose either to use a cache or call the builder every time a new child is needed, but not both.
We can even add an argument ```useCache``` to let users decide that, but I think that can be save for later.

Reply:
Caching is fine. What I meant is the user might misunderstand this and end up thinking that they can't give another instance of the builder which returns something else later. We should clear up that the 2 things are very different. 

Building CupertinoPicker.builder once with one instance of a builder function that returns something different won't work. That pattern won't work in the rest of the framework either since we have no way of knowing whether the function should return something different. 

Building CupertinoPicker.builder once with one instance of a builder function and then building again with another instance of a builder function that returns something different than what that function returned for the same index previously is totally ok and already works the way you coded now. That's the standard way of modifying things in caching lists elsewhere in the framework.

We just need a minor tweak on the phrasing to make it not seem like per index, the result can never change.